test_json.ts

```diff
@@ -0,0 +1,44 @@
+import { toCpon } from "./src/cpon";
+import { fromJson, toJson } from "./src/json";
+import {makeIMap, makeMap, makeMetaMap, RpcValue, RpcValueWithMetaData, withOffset} from "./src/rpcvalue";
+
+const testJsonRoundTrip = (json: string, expected: RpcValue) => {
+	const parsed = fromJson(json);
+	if (parsed instanceof Error) {
+		throw parsed;
+	}
+	const parsedCpon = toCpon(parsed);
+	const expectedCpon = toCpon(expected);
+	if (parsedCpon !== expectedCpon) {
+		throw new Error(`Assertion error:\nExpected: '${expectedCpon}'\nActual: '${parsedCpon}'`);
+	}
+
+	const serializedJson = toJson(expected);
+	if (serializedJson !== json.replaceAll(' ', '')) {
+		throw new Error(`Assertion error:\nExpected: '${json}'\nActual: '${serializedJson}'`);
+	}
+
+};
+
+const MINUTE = 1;
+const HOUR = 60 * MINUTE;
+
+const tests = [
+	['{}', makeMap({})],
+	['{"null-field": null}', makeMap({"null-field": undefined})],
+	['["!shvType", "Blob", ""]', new Uint8Array().buffer],
+	['["!shvType", "Blob", "6162a1"]', new Uint8Array([0x61, 0x62, 0xa1]).buffer],
+	['["!shvType", "IMap", {}]', makeIMap({})],
+	['["!shvType", "IMap", {"1": 2}]', makeIMap({1: 2})],
+	['["!shvType", "IMap", {"1": 2, "2": "foo"}]', makeIMap({1: 2, 2: "foo"})],
+	['["!shvType", "DateTime", "2021-11-08T01:02:03+05"]', withOffset(new Date("2021-11-08T01:02:03+05:00"), 5 * HOUR)],
+	['["!shvType", "DateTime", "2021-11-08T01:02:03-0815"]', withOffset(new Date("2021-11-08T01:02:03-08:15"), -8 * HOUR - 15 * MINUTE)],
+	['["!shvType", "DateTime", "2021-11-08T01:02:03.456-0815"]', withOffset(new Date("2021-11-08T01:02:03.456-08:15"), -8 * HOUR - 15 * MINUTE)],
+	['["!shvMeta", {"1":2}, 42]', new RpcValueWithMetaData(makeMetaMap({1: 2}), 42)],
```

Comment:
Can we also have a test case for UInt, Double and Decimal?

Reply:
Decision is not to have unit, double and decimal in JS, since they are not native types there. We believe that this simplification will not create unsolvable use cases. UInt will be probably removed in next version of SHV.

Reply:
Ok, then it would be meaningful to remove those from the code.

Reply:
Removed UInt/Double/Decimal serialization to JSON